### PR TITLE
Feature/add transcode

### PR DESCRIPTION
**Pull request by strausr**
#### What does this PR solve?
Added transcode functionality
This pr is missing transcode.toAnimate as on confluence it is marked to be re-opened
JSDOC match php expect for fpsRange which is unknown  

#### Final checklist
- [ ] JSdoc - Add proper docstrings based on our PHP implementation
- [X] JSdoc - Hide private functions using @private
- [X] Implementation is aligned to Spec.
- [X] Tests - Add proper tests to the added code

### CODE REVIEW

File: src/actions/transcode/Transcode.ts [hunk 1]
```diff
@@ -1,12 +1,92 @@
...
+function fps(from: number): FPSAction {
+  return new FPSAction(from);
+}
+
+/**
+ * @memberOf Actions.Transcode
+ * @description Controls the range of acceptable FPS (Frames Per Second) to ensure that video (even when optimized) is
+ * delivered with an expected FPS level (helps with sync to audio).
+ * @param from frame rate
+ * @param to frame rate
```
Comment (patrick-tolosa) [inline src/actions/transcode/Transcode.ts:63]: missing types in JSDoc

File: src/actions/transcode/StreamingProfile.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+class StreamingProfileAction extends Action {
+  private profile: string|number;
+
+  constructor(profile: string|number) {
+    super();
+    this.profile = profile;
+  }
+
+  protected prepareParam(): this {
+    this.addParam(new Param('sp', this.profile));
```
Comment (patrick-tolosa) [inline src/actions/transcode/StreamingProfile.ts:13]: shouldn't this just be in the constructor?
Do we need it in the prepare param?

**Review by patrick-tolosa — APPROVED** [review]

File: src/actions/transcode/StreamingProfile.ts (hunk 2 above)
Comment (strausr) [reply src/actions/transcode/StreamingProfile.ts:13]: updated

File: src/actions/transcode/Transcode.ts (hunk 1 above)
Comment (strausr) [reply src/actions/transcode/Transcode.ts:63]: mentioned in the comments its missing

**Pull request merged**